doc/userguide/rules/http-keywords.rst

```diff
@@ -1,180 +1,133 @@
 
-All previous described keywords can be used in combination with a
-buffer in a signature. The keywords ``distance`` and ``within`` are
-relative modifiers, so they may only be used within the same
-buffer. You can not relate content matches against different buffers
-with relative modifiers.
+Example HTTP Response::
+
+  HTTP/1.1 200 OK
+  Content-Type: text/html
+  Content-Length: 258
+  Date: Thu, 14 Dec 2023 20:22:41 GMT
+  Server: nginx/0.8.54
+  Connection: Close
+
+Example signature that would alert on the above response.
+
+.. container:: example-rule
+
+  alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP Stat Code Example"; \
+  flow:established,to_server; :example-rule-options:`http.stat_code; \
+  content:"200"; bsize:8; http.content_type; content:"text/html"; bsize:9;` \
+  classtype:bad-unknown; sid:30; rev:1;)
+
+Request Keywords:
+ * :ref:`http.uri`
+ * :ref:`http.uri.raw`
+ * :ref:`http.method`
+ * :ref:`http.request_line`
+ * :ref:`http.request_body`
+ * :ref:`http.cookie`
+ * :ref:`http.user_agent`
+ * :ref:`http.host`
+ * :ref:`http.host.raw`
+ * :ref:`http.accept`
+ * :ref:`http.accept_lang`
+ * :ref:`http.accept_enc`
+ * :ref:`http.referer`
+ * :ref:`file.name`
+
+Response Keywords:
+ * :ref:`http.stat_msg`
+ * :ref:`http.stat_code`
+ * :ref:`http.response_line`
+ * :ref:`http.response_body`
+ * :ref:`http.server`
+ * :ref:`http.location`
+
+Request or Response Keywords:
+ * :ref:`file.data`
+ * :ref:`http.content_type`
+ * :ref:`http.content_len`
+ * :ref:`http.start`
+ * :ref:`http.protocol`
+ * :ref:`http.header_names`
+ * :ref:`http.header`
+ * :ref:`http.header.raw`
+ * :ref:`http.cookie`
+
+.. _http.method:
 
 http.method
 -----------
 
-With the ``http.method`` sticky buffer, it is possible to match
-specifically and only on the HTTP method buffer. The keyword can be
-used in combination with all previously mentioned content modifiers
-such as: ``depth``, ``distance``, ``offset``, ``nocase`` and ``within``.
+The ``http.method`` keyword matches on the method/verb used in an HTTP request.
+HTTP request methods can be any of the following:
 
-Examples of methods are: **GET**, **POST**, **PUT**, **HEAD**,
-**DELETE**, **TRACE**, **OPTIONS**, **CONNECT** and **PATCH**.
+* GET
+* POST
+* HEAD
+* OPTIONS
+* PUT
+* DELETE
+* TRACE
+* CONNECT
+* PATCH
 
-Example of a method in a HTTP request:
+It is possible to use any of the :doc:`payload-keywords` with the ``http.method`` keyword.
 
-.. image:: http-keywords/method2.png
+Example HTTP Request::
 
-Example of the purpose of method:
+  GET /index.html HTTP/1.1
+  User-Agent: Mozilla/5.0
+  Host: suricata.io
 
-.. image:: http-keywords/method.png
+.. container:: example-rule
 
-.. image:: http-keywords/Legenda_rules.png
+  alert http $HOME_NET any -> $EXTERNAL_NET any (msg:"HTTP Request Example"; \
+  flow:established,to_server; :example-rule-options:`http.method; \
+  content:"GET";`classtype:bad-unknown; sid:2; rev:1;)
 
-.. image:: http-keywords/method1.png
```

Comment:
Could you please explain the removal of some of the images, but not all of them? (IMHO a general explanation could be included in the dedicated commit for removing elements from this section).

Reply:
Ah, I will be removing probably all the images but I only have reworked the HTTP primer, http keywords list and http.method to give an idea of the changes and general formatting